src/dist/replication/lib/bulk_load/replica_bulk_loader.cpp

```diff
@@ -447,6 +454,59 @@ void replica_bulk_loader::check_ingestion_finish()
     }
 }
 
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica_bulk_loader::handle_bulk_load_succeed()
+{
+    // generate checkpoint
+    _replica->init_checkpoint(true);
+
+    _replica->_app->set_ingestion_status(ingestion_status::IS_INVALID);
+    _status = bulk_load_status::BLS_SUCCEED;
+    // TODO(heyuchen): add perf-counter
+}
+
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica_bulk_loader::handle_bulk_load_finish(bulk_load_status::type new_status)
+{
+    if (is_cleanuped()) {
+        ddebug_replica("bulk load states have been cleaned up");
+        return;
+    }
+
+    if (status() == partition_status::PS_PRIMARY) {
+        // TODO(heyuchen): cleanup _primary_states.secondary_bulk_load_states
+    }
+
+    ddebug_replica("bulk load finished, old_status = {}, new_status = {}",
+                   enum_to_string(_status),
+                   enum_to_string(new_status));
+
+    // remove local bulk load dir
+    std::string bulk_load_dir = utils::filesystem::path_combine(
+        _replica->_dir, bulk_load_constant::BULK_LOAD_LOCAL_ROOT_DIR);
+    error_code err = remove_local_bulk_load_dir(bulk_load_dir);
+    if (err != ERR_OK) {
+        tasking::enqueue(
+            LPC_REPLICATION_COMMON,
+            &_replica->_tracker,
+            std::bind(&replica_bulk_loader::remove_local_bulk_load_dir, this, bulk_load_dir),
+            get_gpid().thread_hash());
+    }
+
+    clear_bulk_load_states();
+}
+
+// ThreadPool: THREAD_POOL_REPLICATION
+error_code replica_bulk_loader::remove_local_bulk_load_dir(const std::string &bulk_load_dir)
+{
+    if (!utils::filesystem::directory_exists(bulk_load_dir) ||
```

Comment:
Is there any problem if we return `ERR_OK ` when directory is not exist?

Reply:
It is okay